File: practice/mathematics/fundamentals/diwali_lights/diwali_lights.cpp
#include <iostream>

using namespace std;

long long pow_mod(long long i_a, long long i_deg, long long i_mod)
{
    long long res = 1;
    while (i_deg > 0)
    {
        if (i_deg & 1 == 1)
        {
            res *= i_a;
            if (res >= i_mod)
            {
                res %= i_mod;
            }
        }
        
        i_a *= i_a;
        if (i_a >= i_mod)
        {
            i_a %= i_mod;
        }
        
        i_deg /= 2;
    }
    
    return res;
}

int main()
{
    int t;
    long long n;
    
    cin >> t;
    while (t--)
    {
        cin >> n;
        long long res = pow_mod(2, n, 100000) - 1;
        cout << res << endl;
    }
    return 0;
}

File: practice/mathematics/fundamentals/diwali_lights/diwali_lights.hs
module Main where

pow_mod :: Integer -> Integer -> Integer -> Integer -> Integer
pow_mod a 0 m res = res
pow_mod a deg m res
    | deg `mod` 2 == 1 = pow_mod (a * a `mod` m) (deg `div` 2) m (res * a `mod` m)
    | otherwise        = pow_mod (a * a `mod` m) (deg `div` 2) m res

solve :: Integer -> Integer
solve n = pow_mod 2 n 100000 1 - 1

main :: IO ()
main = do
    t <- readLn :: IO Int
    d <- getContents
    putStr $ unlines $ map (show . solve . (read :: String -> Integer)) $ lines d

